### PR TITLE
AuthenticationService class 

### PR DESCRIPTION
**Pull request by MXPOL**
Created Authentication class in services folder, this class responsible for all things related to authentication including auth routes (login, callback, logout) and also authentication middleware that checks if the user is logged in and if no it will redirect the user to the login screen.

Pros of this change: `velo-external-db` module doesn't have any package that related to authentication

### CODE REVIEW

File: packages/external-db-authorization/lib/factory.js [hunk 1]
```diff
@@ -1,37 +1,33 @@
...
 }
 
-const createAuthService = async(vendor, config) => {
+const initAuthProvider = async(vendor, config) => {
 
   const { auth: authConfig } = await config.readConfig()
-  const { validAuthConfig } = await config.configStatus()
+  const authInformation = await config.configStatus()
+
+  const authProvider = authProviderFor(vendor, authConfig)
```
Comment (Idokah) [inline packages/external-db-authorization/lib/factory.js:27]: If the case was default (local/azure) , so you receive an object back from that function, how do you handle this response? I don't find it in the code

**Comment by MXPOL** [comment]
@noam-almog  ready for review, can you check it?

File: packages/external-db-authorization/lib/services/authentication_service.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        return this.authRouter
+    }
+
+    createLoginHandler() {
+
+        this.authRouter.use(this.loginPath, passport.authenticate(this.authName))
+    }
+
+    createLoginCallbackHandler() {
+        const redirectOptions = { successReturnToOrRedirect: '/', failureRedirect: '/auth/login' }
```
Comment (Idokah) [inline packages/external-db-authorization/lib/services/authentication_service.js:42]: I'm not sure that it supposed to be in the constructor (maybe for the tests you tell me), 
but have `this.loginPath` in constructor, maybe use it here in `failureRedirect` instead hard-coded?

File: packages/external-db-authorization/package.json [hunk 3]
```diff
@@ -25,11 +25,13 @@
   },
   "dependencies": {
     "@aws-sdk/client-cognito-identity-provider": "^3.47.0",
+    "express": "^4.17.2",
```
Comment (noam-almog) [inline packages/external-db-authorization/package.json:28]: we should not import express here, i thought the security model should include passport here, but not express

Comment (MXPOL) [reply packages/external-db-authorization/package.json:28]: We decided that the authentication will be middleware, therefore I have to import Express and use it in the packge ..

Comment (noam-almog) [reply packages/external-db-authorization/package.json:28]: middleware is related to express therefor should be on the main module, the api's being called in it should be on the authorization module
(and change the module to security on another PR)

**Pull request closed without merging**